src/main/java/com/gudimov/movieland/dao/cache/GenreCache.java

```diff
@@ -0,0 +1,56 @@
+package com.gudimov.movieland.dao.cache;
+
+import com.gudimov.movieland.dao.GenreDao;
+import com.gudimov.movieland.dao.jdbc.JdbcGenreDao;
+import com.gudimov.movieland.entity.Genre;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Primary;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Repository;
+
+import javax.annotation.PostConstruct;
+import java.util.*;
+
+@Repository()
+@Primary
+public class GenreCache implements GenreDao {
+    private final Logger LOG = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    JdbcGenreDao jdbcGenreDao;
+
+    private volatile List<Genre> genreCacheList;
+
+    public List<Genre> getAll() {
+        LOG.info("Start genre cache get all");
+        ArrayList<Genre> genresCopy = new ArrayList<>();
+        for (Genre genre : genreCacheList) {
```

Comment:
ConcurrentModificationException. You're iterating on the collection, which can be changed in invalidate() method during scheduling.